src/components/modals/EditTemplateModal.tsx

```diff
@@ -0,0 +1,240 @@
+import { BasicRoundedButton } from "@/components/buttons/basic-rounded-button/Basic-rounded-button";
+import { Exercise } from "@/lib/exercises/exercise";
+import { ExercisesDictionary } from "@/lib/exercises/exercises-dictionary";
+import { ExerciseActivity } from "@/models/exercise-activity.model";
+import { Log } from "@/models/log.model";
+import { Set } from "@/models/set.model";
+import { Modal } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { FiPlus, FiSearch, FiX } from "react-icons/fi";
+
+interface EditTemplateModalProps {
+  open: boolean;
+  onClose: () => void;
+  templateData: ExerciseActivity[] | null;
+  templateNameData: string | null;
+  onUpdateTemplate: (updatedTemplateData: Partial<Log>) => void;
+  templateId: string | null;
+}
+
+const EditTemplateModal: React.FC<EditTemplateModalProps> = ({
+  open,
+  onClose,
+  templateData,
+  templateNameData,
+  onUpdateTemplate,
+  templateId,
+}) => {
+  const [templateName, setTemplateName] = useState<string>("");
+  const [errorMessage, setErrorMessage] = useState<string>("");
+  const [searchInput, setSearchInput] = useState<string>("");
+  const [searchResults, setSearchResults] = useState<Exercise[]>([]);
+  const [selectedExercises, setSelectedExercises] = useState<
+    ExerciseActivity[]
+  >([]);
+  const [toggleSearchBar, setToggleSearchBar] = useState<boolean>(false);
+
+  const exercisesArray = Object.values(ExercisesDictionary);
+
+  useEffect(() => {
+    if (templateData) {
+      setSelectedExercises(templateData);
+    }
+  }, [templateData]);
+
+  useEffect(() => {
+    if (templateNameData) {
+      setTemplateName(templateNameData);
+    }
+  }, [templateNameData]);
+
+  useEffect(() => {
+    if (searchInput) {
+      const results = exercisesArray.filter((exercise) =>
+        exercise.label.toLowerCase().includes(searchInput.toLowerCase())
+      );
+      setSearchResults(results);
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchInput]);
+
+  const handleSelectExercise = (exercise: Exercise): ExerciseActivity => {
+    const newExercise: ExerciseActivity = {
+      exerciseName: exercise.label,
+      sets: [{ setNumber: 1, reps: 0, weight: 0, unit: "lbs" }],
+    };
+
+    setSelectedExercises((prev) => [...prev, newExercise]);
+    setSearchInput("");
+    setSearchResults([]);
+
+    return newExercise;
```

Comment:
🤔I noticed here an event handler was returning a value. Do we need to return a value for this function? Maybe I am missing something?

Reply:
Good catch! There should be no return value, the function is just supposed to update the state with the new exercise and reset the search.